testsuite/src/setupTex.ts

```diff
@@ -34,3 +41,104 @@ export function tex2mml(tex: string) {
   return convert(tex);
 };
 
+// Machinery for measuring the completeness of our macro tests.
+
+const tokens: Map<string, Set<string>> = new Map();
+
+/**
+ * Adds a token to the token set.
+ * @param {string} name Table name.
+ * @param {string} token Token string.
+ */
+function addToken(name: string, token: string) {
+  let tokenSet = tokens.get(name);
+  if (!tokenSet) {
+    tokenSet = new Set();
+    tokens.set(name, tokenSet);
+  }
+  tokenSet.add(token);
+}
+
+/**
+ * Set difference.
+ * @param {Set<string>} exp Expected elements.
+ * @param {Set<string>} act Actual elements.
+ * @return {Set<string>} Expected setminus actual.
+ */
+function setdifference(exp: Set<string>, act: Set<string>): Set<string> {
+  act.forEach(x => exp.delete(x));
+  return exp;
+}
+
+/**
+ * Diff between macros
+ * @param {string} handler
+ * @return {[Set<string>, number, number]}
+ */
+function diffMacros(handler: string): [Set<string>, number, number] {
+  const expected = MapHandler.getMap(handler);
+  if (!expected || expected instanceof RegExpMap) {
+    return [null, 0, 0];
+  }
+  const actual = tokens.get(handler);
+  const expSet = new Set((expected as any).map.keys()) as Set<string>;
+  const expSize = expSet.size;
+  if (!actual) {
+    return [expSet, expSize, 0];
+  }
+  return [setdifference(expSet, actual), expSize, actual.size];
+}
+
+interface tables {
+  table: string,
+  size: number,
+  actual: number,
+  missing: string[]
+}
+
+/**
+ * Gets tested tokens for a configuration and pushes them to the intermediary
+ * test file.
+ *
+ * @param {string} configuration The name of the configuration.
+ */
+export function getTokens(configuration: string) {
+  const config = ConfigurationHandler.get(configuration);
+  if (!config) {
+    console.error(`Something went wrong for configuration ${configuration}`);
+  }
+  const handlers = config[ConfigurationType.HANDLER];
+  // We can omit `handlers[HandlerType.DELIMITER]`.
+  const allHandlers = [].concat(
+    handlers[HandlerType.CHARACTER],
+    handlers[HandlerType.MACRO],
+    handlers[HandlerType.ENVIRONMENT]);
+  let tables: tables[] = [];
+  let outJSON: {
+    configuration: string,
+    tables: tables[]
+  } = {configuration: configuration, tables: tables};
+  allHandlers.forEach((handler) => {
+    const [diff, exp, act] = diffMacros(handler);
+    if (diff) {
+      tables.push({
+        table: handler,
+        size: exp,
+        actual: act,
+        missing: Array.from(diff)
+      })
+    }
+  });
+  fs.appendFile(tmpJsonFile, ',' + JSON.stringify(outJSON, null, 2), (_err) => {
+    console.warn(`could not write file ${tmpJsonFile}`);
+  });
```

Comment:
Should this be using `fs.appendFileSync`?  I got the error message when I was running just a single file, e.g.

``` shell
pnpm -s test testsuite/tests/input/tex/Ams.test.ts
```

and got no table output from there reporter (the error message shows up as a warning above the coverage table).

If I change these three lines to 

``` js
  try {
    fs.appendFileSync(tmpJsonFile, ',' + JSON.stringify(outJSON, null, 2));
  } catch (err) {
    console.warn(`could not write file ${tmpJsonFile}: ${err.message}`);
  };
```

then it works as expected.

Reply:
As an aside, using

``` shell
pnpm -s test -- --collectCoverageFrom 'mjs/input/tex/ams/*.js' testsuite/tests/input/tex/Ams.test.ts
```

limits the coverage table to just the ams files, which makes for a nicer result when trying to test just one extension.

Reply:
Changed to use `appendFileSync` everywhere. I did get the occasional non-deterministic error on this, but rerunning would get rid of it (probably due to the change in test order with the jest cache) but overall async seemed to be faster. But I don't think the speed gain is worth it.